File: cheat_sheet/go/2_extend_feature/internal/domain/organization/organization.go
package organization

import (
	"time"
)

// LanguageCode represents supported language codes.
type LanguageCode string

const (
	LanguageCodeZhHant LanguageCode = "zh-hant"
	LanguageCodeZhHans LanguageCode = "zh-hans"
	LanguageCodeEn     LanguageCode = "en"
	// Add more language codes as needed
)

// Organization represents an organization entity in the system.
// It contains the core business information and settings for an organization.
type Organization struct {
	ID              int          `json:"id"`
	Name            string       `json:"name"`
	UUID            string       `json:"uuid"`
	PlanID          *string      `json:"plan_id,omitempty"`
	EnableTwoFactor bool         `json:"enable_two_factor"`
	Timezone        string       `json:"timezone"`
	LanguageCode    LanguageCode `json:"language_code"`
	Enable          bool         `json:"enable"`
	CreatedAt       time.Time    `json:"created_at"`
	UpdatedAt       time.Time    `json:"updated_at"`
	ExpiredAt       *time.Time   `json:"expired_at,omitempty"`
}

File: cheat_sheet/python/1_rewrite_brownfield/internal/domain/organization/organization.py
"""Organization domain models."""

from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel


class LanguageCode(StrEnum):
    """Language code enumeration."""

    ZH_HANT = "zh-hant"
    ZH_HANS = "zh-hans"
    EN = "en"


class Organization(BaseModel):
    """Organization domain model.

    Represents an organization entity in the system.
    It contains the core business information and settings for an organization.
    """

    id: int
    name: str
    uuid: str
    plan_id: str | None = None
    enable_two_factor: bool
    timezone: str
    language_code: LanguageCode
    enable: bool
    created_at: datetime
    updated_at: datetime
    expired_at: datetime | None = None

    class Config:
        """Pydantic configuration."""

        use_enum_values = True
